lib/ansible/parsing/plugin_docs.py

```diff
@@ -68,11 +68,15 @@ def read_docstring(filename, verbose=True, ignore_errors=True):
         # Metadata is per-file and a dict rather than per-plugin/function and yaml
         data['metadata'] = extract_metadata(module_ast=M)[0]
 
-        # remove version
         if data['metadata']:
+            # remove version
             for x in ('version', 'metadata_version'):
                 if x in data['metadata']:
                     del data['metadata'][x]
+        else:
+            # Add default metadata
+            data['metadata'] = {'supported_by': 'community',
+                                'status': ['preview']}
```

Comment:
wont this now make metadata optional for modules?

Reply:
yes.  metadata *is* optional for modules.  That's how we designed it.